### DISCUSSION TITLE
Building static sites that doesn't require any web server

### DISCUSSION THREAD

**Discussion by 1grzyb1** [withastro/roadmap#961] [Proposal]
# Summary
Provide a method to generate static HTML files that can be opened directly without requiring a web server.

# Background & Motivation
The goal is to create HTML documentation that clients can open and view directly from their file system without needing to set up or use a web server. This ensures clients have an easy and convenient way to access the documentation.

Currently, generated Astro projects require a web server to function correctly, making it inconvenient for clients who need to simply open an index.html file and browse the documentation.

# Goals
Generate Self-Contained HTML: Ensure the HTML files contain all necessary resources (CSS, JavaScript, images, etc.) embedded or referenced in a way that works without a web server.
Simplify Client Access: Enable clients to open the index.html file directly from their file system without any additional setup.
Maintain Quality: Ensure the generated documentation maintains high visual and functional quality.

**Comment by florian-lefebvre** [comment]
I think an option would be to accept `'.'` as a base to say "use relative links"

**Reply by 1grzyb1** [reply]
Currently setting '.' as base will generates '/./'.

The second issue is with JavaScript modules. When importing them in a browser without a web server it gives a CORS error

**Reply by florian-lefebvre** [reply]
Ah interesting I didn't know that

**Reply by 1grzyb1** [reply]
So what do you think about such feature? Is it even possible to add to astro?

**Reply by florian-lefebvre** [reply]
I'm not familiar with this part of Astro but sounds like it would require quite some work, and it's unlikely this gets tackled any time soon. We measure interest based on the upvotes so if we see there's a clear demand, this can move to Stage 2

**Reply by Fryuni** [reply]
There is a community integration with this idea: https://www.npmjs.com/package/astro-relative-links

Simple tests worked fine, but I haven't looked at the code
